lib/json/common.rb

```diff
@@ -179,6 +179,20 @@ def parse!(source, opts = {})
     Parser.new(source, opts).parse
   end
 
+
+  # Parses the content of a file
+  # See parse method documentation for more information.
+  def load_file(filespec, opts = {})
+    parse(File.read(filespec))
```

Comment:
@keithrbennett I'm just wondering, shouldn't we pass options for 'File.read'? It might be useful for supporting different encodings, etc. It's not related but I have a case where I struggle with BOM character in the beginning of the CSV file and by passing encoding I can simply remove those additional bytes. Maybe that would work similarly in JSON.

Reply:
First I want to thank you for pointing out to me that I was failing to pass the passed `opts` along to the called methods. I've fixed that and pushed the fix.

Regarding your point about `File.read` options, do I understand you correctly that you would want to specify options to `File.read`, not `JSON.parse`? If so, I'm not sure how we could do that, since we are already using the Ruby hash last parameter feature for the `JSON.parse` options.

Any strategy to handle options to `File.read` would need to be balanced with the cost of the complexity added to support it. Can you suggest a way to accommodate this need in the least complex way?

Reply:
Yes, but I'm happy with current solution. I don't have idea how to solve it in a elegant way. Maybe maintainer can help. And sorry for the oftop, anyway it's better than before :)

Reply:
Best not mix options. If a user need options, they can do the `File.read`. FWIW, `YAML.load_file` has no file option either.